scipy/stats/_stats_py.py

```diff
@@ -9691,9 +9785,59 @@ def wasserstein_distance(u_values, v_values, u_weights=None, v_weights=None):
     >>> wasserstein_distance([3.4, 3.9, 7.5, 7.8], [4.5, 1.4],
     ...                      [1.4, 0.9, 3.1, 7.2], [3.2, 3.5])
     4.0781331438047861
-
+    >>> wasserstein_distance([[0, 2, 3], [1, 2, 5]], [[3, 2, 3], [4, 2, 5]])
+    3.0
+    >>> wasserstein_distance([[0, 2.75], [2, 209.3], [0, 0]],
+    ...                      [[0.2, 0.322], [4.5, 25.1808]],
+    ...                      [0.4, 5.2, 0.114], [0.8, 1.5])
+    174.15840245217169
     """
-    return _cdf_distance(1, u_values, v_values, u_weights, v_weights)
+    m, n = len(u_values), len(v_values)
+    u_values = asarray(u_values).reshape(m, -1)
+    v_values = asarray(v_values).reshape(n, -1)
+
+    # if data is 1D then call the cdf_distance function
+    if u_values.shape[1] == 1 and v_values.shape[1] == 1:
+        return _cdf_distance(1,
+                             u_values.ravel(), v_values.ravel(),
+                             u_weights, v_weights)
+
+    u_values, u_weights = _validate_distribution(u_values, u_weights)
+    v_values, v_weights = _validate_distribution(v_values, v_weights)
+
+    # if dimensions are not equal throw error
+    if u_values.shape[1] != v_values.shape[1]:
+        raise ValueError('Invalid input values. Please provide two numpy'
+                         'arrays with either both one-dimensional or both'
+                         'two-dimensional arrays with the same number of'
+                         'columns.')
```

Comment:
At this point, we know that the user was not trying to pass one-dimensional arrays. Also, it's ambiguous whether the arrays must have the same number of columns if they're one-dimensional.
```suggestion
        raise ValueError('Invalid input values. If two-dimensional,'
                         '`u_values` and `v_values` must have the same'
                         'number of'columns.')
```